src/CommonDBTM.php

```diff
@@ -5498,6 +5498,8 @@ public function addFiles(array $input, $options = [])
                 $is_recursive = $input['is_recursive'];
             } else if (isset($input['_job']->fields['is_recursive'])) {
                 $is_recursive = $input['_job']->fields['is_recursive'];
+            } else if ($this->getType() == KnowbaseItem::getType()) {
+                $is_recursive = 1;
```

Comment:
```suggestion
            } else if ($this instanceof CommonDBVisible) {
                // CommonDBVisible visibility restriction is unpredictable as
                // it may change over time, and can be related to dynamic profiles assignation.
                // Related documents have to be available on all entities.
                $is_recursive = 1;
```